crates/cdk/src/mint/mint_20.rs

```diff
@@ -0,0 +1,102 @@
+use tracing::instrument;
+
+use super::nut20::{MintQuoteBolt12Request, MintQuoteBolt12Response};
+use super::{Mint, MintQuote, PaymentMethod};
+use crate::util::unix_time;
+use crate::{Amount, Error};
+
+impl Mint {
+    /// Create new mint bolt11 quote
+    #[instrument(skip_all)]
+    pub async fn get_mint_bolt12_quote(
+        &self,
+        mint_quote_request: MintQuoteBolt12Request,
+    ) -> Result<MintQuoteBolt12Response, Error> {
+        let MintQuoteBolt12Request {
+            amount,
+            unit,
+            description,
+            single_use,
+            expiry,
+            pubkey,
+        } = mint_quote_request;
+
+        let nut18 = &self
+            .mint_info
+            .nuts
+            .nut18
+            .as_ref()
+            .ok_or(Error::UnsupportedUnit)?;
+
+        if nut18.disabled {
+            return Err(Error::MintingDisabled);
+        }
+
+        let ln = self.bolt12_backends.get(&unit).ok_or_else(|| {
+            tracing::info!("Bolt11 mint request for unsupported unit");
```

Comment:
can we change this to reflect Bolt12 -> ` Bolt12 mint request for unsupported unit`